src/styles/blocks/grid.scss

```
@@ -0,0 +1,53 @@
+.grid {
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  column-gap: 20px;
+
+  @include onTablet {
+    grid-template-columns: repeat(6, 1fr);
+    column-gap: 24px;
+  }
+
+  @include onDesktop {
+    grid-template-columns: repeat(12, 1fr);
+    column-gap: 24px;
+  }
+
+  &--tablet {
+    display: block;
+
+    @include onTablet {
+      display: grid;
+    }
+  }
+
+  &--desktop {
+    display: block;
+
+    @include onDesktop {
+      display: grid;
+    }
+  }
+
+  &__item {
+    @for $start from 1 through 12 {
+      @for $end from $start through 12 {
+        &--#{$start}-#{$end} {
+          grid-column: #{$start} / #{$end + 1};
+        }
+
+        &--tablet-#{$start}-#{$end} {
+          @include onTablet {
+            grid-column: #{$start} / #{$end + 1};
+          }
+        }
+
+        &--desktop-#{$start}-#{$end} {
+          @include onDesktop {
+            grid-column: #{$start} / #{$end + 1};
+          }
+        }
+      }
+    }
```

the issue is in this loop. create 3 separate loops for each classname
```suggestion
@for $start from 1 through 12 {
      @for $end from $start through 12 {
        &--#{$start}-#{$end} {
          grid-column: #{$start} / #{$end + 1};
        }
      }
    }
    
    @for $start from 1 through 12 {
      @for $end from $start through 12 {
        &--tablet-#{$start}-#{$end} {
          @include onTablet {
            grid-column: #{$start} / #{$end + 1};
          }
        }
      }
    }
    
    @for $start from 1 through 12 {
      @for $end from $start through 12 {
        &--desktop-#{$start}-#{$end} {
          @include onDesktop {
            grid-column: #{$start} / #{$end + 1};
          }
        }
      }
    }
```

---

Thank you
Exactly!